lib/transport/abstract-transport.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+var util = require("util");
+var EventEmitter = require('events');
+
+var AbstractTransport = function () {
+	if (this.constructor === AbstractTransport) {
+		throwBecauseAbstract();
```

Comment:
Well what I thought was particularly clever about your initial implementation was unifying the "interfaces" of the two existing transports, hence the idea for an `AbstractTransport class. Also, it's incredibly likely that we will be able to make use of the `AbstractTransport` for unit testing.

As far as exporting it goes: I'm not sure it makes sense to require the entirety of the `amqp10` module simply to implement a custom transport. Duck-typing may feel unsafe/funny/whatever, but it's sort of the law of the land in javascript/node world (perhaps a look at other similar cases would be illuminating here, maybe sails' waterline db backend or something as its highly modular).

I'm not too tied to any of it, just dropping my two cents

I'm amenable to either approach, I just felt that using the abstract for internal transports buys us some consistency/reliability cheaply.